ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -162,77 +157,77 @@ S3Path(const std::string &dir, const std::string &name)
-    *exist = S3CacheExists(fileSystem, name);
-    if (*exist)
+    // Check if file exists in the cache.
+    *fileExists = S3CacheExists(fileSystem, name);
+    if (*fileExists) {
+        s3->log->LogDebugMessage("S3FileExists: Found file in cache.");
         return (ret);
+    }
 
-    /* It's not in the cache, try the S3 bucket. */
+    // It's not in the cache, try the S3 bucket.
+    size_t objectSize;
     s3->statistics.objectExistsCount++;
-    if ((ret = fs->connection->ObjectExists(name, *exist, objectSize)) != 0)
-        s3->log->LogErrorMessage("S3Exist: ObjectExists request to S3 failed.");
+    if ((ret = fs->connection->ObjectExists(name, *fileExists, objectSize)) != 0)
+        s3->log->LogErrorMessage("S3FileExists: ObjectExists request to S3 failed.");
+    else
+        s3->log->LogDebugMessage("S3FileExists: Found file in S3.");
 
     return (ret);
 }
 
-/*
- * S3CacheExists --
- *     Checks whether the given file exists in the cache.
- */
+// Checks whether the given file exists in the cache.
 static bool
 S3CacheExists(WT_FILE_SYSTEM *fileSystem, const std::string &name)
 {
-    const std::string path = S3Path(((S3_FILE_SYSTEM *)fileSystem)->cacheDir, name);
+    const std::string path = S3Path(((S3FileSystem *)fileSystem)->cacheDir, name);
     return (LocalFileExists(path));
 }
 
-/*
- * LocalFileExists --
- *     Checks whether a file corresponding to the provided path exists locally.
- */
+// Checks whether a file corresponding to the provided path exists locally.
 static bool
 LocalFileExists(const std::string &path)
 {
     std::ifstream f(path);
     return (f.good());
 }
 
-/*
- * S3GetDirectory --
- *     Return a copy of a directory name after verifying that it is a directory.
- */
+// Return a copy of a directory name after verifying that it is a directory.
 static int
-S3GetDirectory(const S3_STORAGE &s3, const std::string &home, const std::string &name, bool create,
+S3GetDirectory(const S3Storage &s3, const std::string &home, const std::string &name, bool create,
   std::string &copy)
 {
+    // copy must be defined before the function returns.
     copy = "";
 
     struct stat sb;
     int ret;
     std::string dirName;
 
-    /* For relative pathnames, the path is considered to be relative to the home directory. */
+    // For relative pathnames, the path is considered to be relative to the home directory.
     if (name[0] == '/')
         dirName = name;
     else
         dirName = home + "/" + name;
 
     ret = stat(dirName.c_str(), &sb);
     if (ret != 0 && errno == ENOENT && create) {
-        mkdir(dirName.c_str(), 0777);
+        try {
+            std::filesystem::create_directory(dirName.c_str());
+            std::filesystem::permissions(dirName.c_str(), std::filesystem::perms::all);
+        } catch (std::filesystem::filesystem_error const &e) {
+            s3.log->LogErrorMessage(std::string("S3GetDirectory: ") + e.what());
+        }
+
         ret = stat(dirName.c_str(), &sb);
```

Comment:
Should stat be replaced with something like ```std::filesystem::file_status```

Reply:
I've replaced stat with std::filesystem::status.